app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/LocalFileTransferActivity.java

```diff
@@ -0,0 +1,437 @@
+  private static final int PERMISSION_REQUEST_CODE_STORAGE_WRITE_ACCESS = 2;
+
+  @Inject SharedPreferenceUtil sharedPreferenceUtil;
+  @Inject AlertDialogShower alertDialogShower;
+
+  @BindView(R.id.toolbar_local_file_transfer) Toolbar actionBar;
+  @BindView(R.id.text_view_device_name) TextView deviceName;
+  @BindView(R.id.progress_bar_searching_peers) ProgressBar searchingPeersProgressBar;
+  @BindView(R.id.list_peer_devices) ListView listViewPeerDevices;
+  @BindView(R.id.text_view_empty_peer_list) TextView textViewPeerDevices;
+  @BindView(R.id.recycler_view_transfer_files) RecyclerView filesRecyclerView;
+
+  private boolean isFileSender = false;    // Whether the device is the file sender or not
+
+  private @NonNull WifiDirectManager wifiDirectManager = new WifiDirectManager(this);
+
+  private ArrayList<FileItem> filesForTransfer = new ArrayList<>();
+  private FileListAdapter fileListAdapter;
+
+  private List<WifiP2pDevice> availablePeerDevices = new ArrayList<WifiP2pDevice>();
+  private boolean hasSenderStartedConnection = false;
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    KiwixApplication.getApplicationComponent().activityComponent()
+        .activity(this)
+        .build()
+        .inject(this);
+    setTheme(sharedPreferenceUtil.nightMode() ? R.style.Theme_AppCompat_DayNight_NoActionBar : R.style.AppTheme);
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_local_file_transfer);
+    ButterKnife.bind(this);
+
+    /*
+     * Presence of file Uris decides whether the device with the activity open is a sender or receiver:
+     * - On the sender device, this activity is started from the app chooser post selection
+     * of files to share in the Library
+     * - On the receiver device, the activity is started directly from within the 'Get Content'
+     * activity, without any file Uris
+     * */
+    Intent filesIntent = getIntent();
+    ArrayList<Uri> fileUriArrayList;
+    fileUriArrayList = filesIntent.getParcelableArrayListExtra(Intent.EXTRA_STREAM);
+    isFileSender = (fileUriArrayList != null && fileUriArrayList.size() > 0);
+
+    setSupportActionBar(actionBar);
+    actionBar.setNavigationIcon(R.drawable.ic_close_white_24dp);
+    actionBar.setNavigationOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        finish();
+      }
+    });
+
+    listViewPeerDevices.setAdapter(
+        new WifiPeerListAdapter(this, R.layout.row_peer_device, availablePeerDevices));
+
+    if (isFileSender) {
+      for (int i = 0; i < fileUriArrayList.size(); i++) {
+        filesForTransfer.add(new FileItem(getFileName(fileUriArrayList.get(i)), TO_BE_SENT));
+      }
+
+      displayFileTransferProgress(filesForTransfer);
+    }
+
+    wifiDirectManager.createWifiDirectManager(sharedPreferenceUtil, alertDialogShower, filesForTransfer, fileUriArrayList);
```

Comment:
`create` is the wrong word here, I would just call this method `start`. Also it probably does not need sharedPref/dialogShower. Those can be provided to it's constructor when you inject it, if the DI code is getting a bit complicated then DM me.
Also sending the uris and the fileitems feels wrong, you could extend FileItems to contain the uris and only deal in FileItems. You could add another constructor to FileItem that describes this
```
public FileItem(Uri fileUri){
this(fileUri, getFilename(fileUri), TO_BE_SENT);
}
```

Reply:
I've made the changes to [FileItem](https://github.com/kiwix/kiwix-android/pull/1276/commits/d6cc0b8faa2b79918fce36f2da381c55e4787770), and changed the WDM method name to [start/stop](https://github.com/kiwix/kiwix-android/pull/1276/commits/d6cc0b8faa2b79918fce36f2da381c55e4787770#diff-e1d59ae6e15dcb502e125fb6e243d15dR133).
Still trying to figure out how exactly DI is to be incorporated, I'll DM if I get stuck after I'm done learning it